sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProtoTest.java

```diff
@@ -282,28 +487,36 @@ public void testNestedFromTableSchema() {
           .put("arrayvalue", ImmutableList.of("hello", "goodbye"))
           .build();
 
-  private void assertBaseRecord(DynamicMessage msg) {
+  private void assertBaseRecord(DynamicMessage msg, boolean withF) {
     Map<String, Object> recordFields =
         msg.getAllFields().entrySet().stream()
             .collect(
                 Collectors.toMap(entry -> entry.getKey().getName(), entry -> entry.getValue()));
-    assertEquals(BASE_ROW_EXPECTED_PROTO_VALUES, recordFields);
+    assertEquals(
+        withF ? BASE_ROW_EXPECTED_PROTO_VALUES : BASE_ROW_NO_F_EXPECTED_PROTO_VALUES, recordFields);
   }
 
   @Test
   public void testMessageFromTableRow() throws Exception {
     TableRow tableRow =
-        new TableRow().set("nestedValue1", BASE_TABLE_ROW).set("nestedValue2", BASE_TABLE_ROW);
+        new TableRow()
+            .set("nestedValue1", BASE_TABLE_ROW)
```

Comment:
Do we also support fields named "f" at top level (not just nested or repeated fields) after this change ? If let's add a test for that as well. If not let's clarify in the documentation.

Reply:
Done